metricflow/inference/rule/base.py

```diff
@@ -1,123 +1,11 @@
 from __future__ import annotations
 from abc import ABC, abstractmethod
+from typing import List
 
-from dataclasses import dataclass
```

Comment:
These were moved to `metricflow/inference/models.py` since they're common classes to all inference steps and don't really belong to `rules` only.